cli/cmd/install/install.go

```diff
@@ -0,0 +1,456 @@
+	flagSetValues       []string
+	flagFileValues      []string
+
+	flagKubeConfig  string
+	flagKubeContext string
+
+	once sync.Once
+	help string
+}
+
+func (c *Command) init() {
+	// Store all the possible preset values in 'presetList'. Printed in the help message.
+	var presetList []string
+	for name := range presets {
+		presetList = append(presetList, name)
+	}
+
+	c.set = flag.NewSets()
+	{
+		f := c.set.NewSet("Command Options")
+		f.BoolVar(&flag.BoolVar{
+			Name:    FlagSkipConfirm,
+			Target:  &c.flagSkipConfirm,
+			Default: DefaultSkipConfirm,
+			Usage:   "Skip confirmation prompt.",
+		})
+		f.BoolVar(&flag.BoolVar{
+			Name:    FlagDryRun,
+			Target:  &c.flagDryRun,
+			Default: DefaultDryRun,
+			Usage:   "Validate installation and return summary of installation.",
+		})
+		f.StringSliceVar(&flag.StringSliceVar{
+			Name:    FlagValueFiles,
+			Aliases: []string{"f"},
+			Target:  &c.flagValueFiles,
+			Usage:   "Path to a file to customize the installation, such as Consul Helm chart values file. Can be specified multiple times.",
+		})
+		f.StringVar(&flag.StringVar{
+			Name:    FlagReleaseName,
+			Target:  &c.flagReleaseName,
+			Default: DefaultReleaseName,
+			Usage:   "Name of the installation. This will be prefixed to resources installed on the cluster.",
+		})
+		f.StringVar(&flag.StringVar{
+			Name:    FlagNamespace,
+			Target:  &c.flagNamespace,
+			Default: DefaultNamespace,
+			Usage:   fmt.Sprintf("Namespace for the Consul installation. Defaults to \"%q\".", DefaultNamespace),
+		})
+		f.StringVar(&flag.StringVar{
+			Name:    FlagPreset,
+			Target:  &c.flagPreset,
+			Default: DefaultPreset,
+			Usage:   fmt.Sprintf("Use an installation preset, one of %s. Defaults to \"%q\"", strings.Join(presetList, ", "), DefaultPreset),
+		})
+		f.StringSliceVar(&flag.StringSliceVar{
+			Name:   FlagSetValues,
+			Target: &c.flagSetValues,
+			Usage:  "Set a value to customize. Can be specified multiple times. Supports Consul Helm chart values.",
+		})
+		f.StringSliceVar(&flag.StringSliceVar{
+			Name:   FlagFileValues,
+			Target: &c.flagFileValues,
+			Usage: "Set a value to customize via a file. The contents of the file will be set as the value. Can be " +
+				"specified multiple times. Supports Consul Helm chart values.",
+		})
+		f.StringSliceVar(&flag.StringSliceVar{
+			Name:   FlagSetStringValues,
+			Target: &c.flagSetStringValues,
+			Usage:  "Set a string value to customize. Can be specified multiple times. Supports Consul Helm chart values.",
+		})
+
+		f = c.set.NewSet("Global Options")
+		f.StringVar(&flag.StringVar{
+			Name:    "kubeconfig",
+			Aliases: []string{"c"},
+			Target:  &c.flagKubeConfig,
+			Default: "",
+			Usage:   "Path to kubeconfig file.",
+		})
+		f.StringVar(&flag.StringVar{
+			Name:    "context",
+			Target:  &c.flagKubeContext,
+			Default: "",
+			Usage:   "Kubernetes context to use.",
+		})
+	}
+
+	c.help = c.set.Help()
+}
+
+func (c *Command) Run(args []string) int {
```

Comment:
@t-eckert's comment: I'm a little concerned about the length of this function. It was hard for me to keep in mind all of the context I needed as I went through it. I think it would make more sense to break the function down into several functions that are executed by the Run function.

Reply:
Broke this up a bit, now only actions pertaining to the main install are in the Run function.